js/ui/grid_core/ui.grid_core.header_filter_core.js

```diff
@@ -360,6 +357,28 @@ export const HeaderFilterView = modules.View.inherit({
         }
     },
 
+    _normalizeHeaderFilterOptions(options) {
```

Comment:
Suggestion: Try to make this function pure and move it to the utils part.
 Use it in place in _getSearchExpr and _initializeListContainer instead of passing the result of it as parameter.